frontend/lib/parsing/ParserContextImpl.h

```diff
@@ -986,16 +986,14 @@ CommentsAndStmt ParserContext::buildFunctionDecl(YYLTYPE location,
 
   // Detect primary methods and create a receiver for them
   bool primaryMethod = false;
-  auto scope = currentScope();
   if (currentScopeIsAggregate()) {
```

Comment:
If secondary and tertiary methods still store `foo` in the `typeExpression` of the `this` formal, wouldn't they run into the same problem that primary methods are having? What happens in that case if there's a field named `foo`? [edit: I'm dumb, it's not a problem cause the class is not our lexical parent and thus the fields are not in scope.]